### DISCUSSION TITLE
SimpleConnectionPool

### DISCUSSION THREAD

**Discussion by DzigaV** [psycopg/psycopg#26] [Ideas]
Hi Daniele, will SimpleConnectionPool make it across from psycopg2? We'll need this to properly utilize async functions.

**Comment by dvarrazzo** [comment]
I'm hoping for a smarter connection pool than psycopg2's... :eyes:

**Reply by DzigaV** [reply]
I've never used it, just aiopg's pool 🙂

But I mentioned SimpleConnectionPool because the the thread-safe pool isn't necessary with async support.

**Comment by dvarrazzo** [comment]
psycopg2 pool started as a Zope thing, which as far as I understand didn't need thread-safety, that's why thread safety appeared prominently in other implementations.

I am gathering some ideas about the characteristics of the pool I have in mind. The idea would be to have an implementation safe both to threads and green threads, maybe even robust to forking (at least able to recognise that the connections are all broken and need disposal) and an async pool with a similar interface, akin to how all objects in psycopg3 have thin sync/async wrappers over state-keeping generators.
